test/e2e_new/config/aws-dev.yaml

```diff
@@ -0,0 +1,104 @@
+---
+# E2E test scenario using local dev images and manifests built from the source tree for following providers:
+# - cluster-api
+# - bootstrap kubeadm
+# - control-plane kubeadm
+# - aws
+
+# For creating local dev images built from the source tree;
+# - from the CAPI repository root, `make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api` to build the cluster-api, bootstrap kubeadm, control-plane kubeadm provider images.
+# - from the CAPA repository root, `make docker-build REGISTRY=gcr.io/k8s-staging-cluster-api` to build the aws provider image.
+
+# Following env variables should be set before running the provider
+
+# export AWS_REGION=
+# export AWS_SSH_KEY_NAME=
+# export AWS_B64ENCODED_CREDENTIALS=
+
+images:
+  # Use local dev images built source tree;
+  - name: gcr.io/k8s-staging-cluster-api/cluster-api-controller-amd64:dev
+    loadBehavior: mustLoad
+  - name: gcr.io/k8s-staging-cluster-api/kubeadm-bootstrap-controller-amd64:dev
+    loadBehavior: mustLoad
+  - name: gcr.io/k8s-staging-cluster-api/kubeadm-control-plane-controller-amd64:dev
+    loadBehavior: mustLoad
+  - name: gcr.io/k8s-staging-cluster-api/cluster-api-aws-controller-amd64:dev
+    loadBehavior: mustLoad
+  - name: quay.io/jetstack/cert-manager-cainjector:v0.11.0
+    loadBehavior: tryLoad
+  - name: quay.io/jetstack/cert-manager-webhook:v0.11.0
+    loadBehavior: tryLoad
+  - name: quay.io/jetstack/cert-manager-controller:v0.11.0
+    loadBehavior: tryLoad
+
+providers:
+
+  - name: cluster-api
+    type: CoreProvider
+    versions:
+      - name: v0.3.0
+        # Use manifest from source files
+        value: ../../../../cluster-api/config
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
+
+  - name: kubeadm
+    type: BootstrapProvider
+    versions:
+      - name: v0.3.0
+        # Use manifest from source files
+        value: ../../../../cluster-api/bootstrap/kubeadm/config
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
+
+  - name: kubeadm
+    type: ControlPlaneProvider
+    versions:
+      - name: v0.3.0
+        # Use manifest from source files
+        value: ../../../../cluster-api/controlplane/kubeadm/config
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
```

Comment:
Does the e2e test framework allow for the same defaulting to the latest "released" version that `clusterctl` does? If so, then we could rely on that here to avoid requiring most developers from having to keep their source trees in sync to test this provider.

Reply:
I think the value here should be along the lines of
`https://github.com/kubernetes-sigs/cluster-api//config?ref=master`, as we have in CAPV https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/test/e2e/e2e.conf

Reply:
Updated to use the latest released versions